ledger/trie/branch.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package trie
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/hash"
+)
+
+type Branch struct {
+	lChild Node
+	rChild Node
+
+	height uint16
+	hash   hash.Hash
+
+	// dirty marks whether the current hash value of the branch is valid.
+	// If this is set to false, the hash needs to be recomputed.
+	dirty bool
```

Comment:
No worries, it's because the comment is wrong 😄 It is indeed that dirty=true means it needs to be recomputed and not the other way around! haha. I'll update the comment, nice catch :)

Reply:
Cool :)